vmm/src/vm.rs

```diff
@@ -821,17 +830,31 @@ impl Vm {
     pub fn create_hypervisor_vm(
         hypervisor: &Arc<dyn hypervisor::Hypervisor>,
         #[cfg(feature = "tdx")] tdx_enabled: bool,
+        #[cfg(feature = "snp")] snp_enabled: bool,
     ) -> Result<Arc<dyn hypervisor::Vm>> {
         hypervisor.check_required_extensions().unwrap();
 
-        // 0 for KVM_X86_LEGACY_VM
-        // 1 for KVM_X86_TDX_VM
-        #[cfg(feature = "tdx")]
-        let vm = hypervisor
-            .create_vm_with_type(u64::from(tdx_enabled))
-            .unwrap();
-        #[cfg(not(feature = "tdx"))]
-        let vm = hypervisor.create_vm().unwrap();
+        cfg_if::cfg_if! {
+            if #[cfg(feature = "tdx")] {
+                let vm = hypervisor
+                    .create_vm_with_type(if tdx_enabled {
+                        2 // KVM_X86_TDX_VM
```

Comment:
I don't thinks this is right. The kvm ioctl expects `vm_type` to be `1` for TD guests.

Reply:
Good catch, I think this got updated here: https://github.com/cloud-hypervisor/cloud-hypervisor/commit/e4ae668bcde2a4409afffae7d85d9d590fcbb10f

My code was based on the older version of this commit.

Reply:
Fixed in the updated PR.